drivers/network/afd/afd/connect.c

```diff
@@ -463,8 +463,66 @@ AfdStreamSocketConnect(PDEVICE_OBJECT DeviceObject, PIRP Irp,
         if( !FCB->RemoteAddress )
             Status = STATUS_NO_MEMORY;
         else
+        {
+	        /* original datagram code set STATUS_SUCCESS and exited here
+            // THIS CODE WAS COPIED FROM STREAM CODE BELOW, IT SHOULD BE SAFE AND 'CORRECT', BUT MAY NOT BE OPTIMAL. */
+            AFD_DbgPrint(MID_TRACE,("AfdStreamSocketConnect: calling WarmSocketConnection(FCB)\n"));
+	        if (FCB->State ==  SOCKET_STATE_CREATED) 
+            { 
+        	    AFD_DbgPrint(MID_TRACE,("AfdStreamSocketConnect: SOCKET_STATE_CREATED is true\n"));
+           	    if (FCB->LocalAddress)
+           	    {
+           		    ExFreePoolWithTag(FCB->LocalAddress, TAG_AFD_TRANSPORT_ADDRESS);
+           	    }
+
+           	    AFD_DbgPrint(MID_TRACE,("AfdStreamSocketConnect: setting FCB->LocalAddress\n"));
+
+           	    FCB->LocalAddress = TaBuildNullTransportAddress(ConnectReq->RemoteAddress.Address[0].AddressType);
+
+           	    if( FCB->LocalAddress ) {
+           		    AFD_DbgPrint(MID_TRACE,("AfdStreamSocketConnect: calling WarmSocketForBind\n"));
+	       	        Status = WarmSocketForBind( FCB, AFD_SHARE_WILDCARD );
+           		    if( NT_SUCCESS(Status) ) {
+           			    AFD_DbgPrint(MID_TRACE,("AfdStreamSocketConnect: setting SOCKET_STATE_BOUND\n"));
+               		    FCB->State = SOCKET_STATE_BOUND;  // <=== send REQUIRES THIS TO USE FCB->RemoteAddress
+	    	        }
+	            }    
+	        }
+
+	        if (FCB->State ==  SOCKET_STATE_BOUND) 
+            { // THIS CODE SECTION EXECUTES AND HANDLES STORING TARGET ADDRESS
+	     	    if (FCB->ConnectReturnInfo)
+	     	    {
+            		AFD_DbgPrint(MID_TRACE,("AfdStreamSocketConnect: ConnectReturnInfo set, dispose allocated obect\n"));
+                	ExFreePoolWithTag(FCB->ConnectReturnInfo, TAG_AFD_TDI_CONNECTION_INFORMATION);
+	         	}
+
+               	AFD_DbgPrint(MID_TRACE,("AfdStreamSocketConnect: calling TdiBuildconnectionInfo with RemoteAddress\n"));
+                Status = TdiBuildConnectionInfo( &FCB->ConnectReturnInfo, &ConnectReq->RemoteAddress );
+
+            	if( NT_SUCCESS(Status) )
+                {
+                   	AFD_DbgPrint(MID_TRACE,("AfdStreamSocketConnect: ConnectReturnInfo placed in FCB\n"));
+                   	if (FCB->ConnectCallInfo)
+                   	{
+            		    AFD_DbgPrint(MID_TRACE,("AfdStreamSocketConnect: ConnectCallInfo set, dispose allocated obect\n"));
```

Comment:
```suggestion
            		    AFD_DbgPrint(MID_TRACE,("AfdStreamSocketConnect: ConnectCallInfo set, dispose allocated object\n"));
```
either that typo-fix or deleting the line altogether.